operator-common/src/main/java/io/strimzi/operator/common/ReconciliationLogger.java

```diff
@@ -0,0 +1,8439 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common;
+
+import java.io.Serializable;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.MessageFactory;
+import org.apache.logging.log4j.spi.AbstractLogger;
+import org.apache.logging.log4j.spi.ExtendedLoggerWrapper;
+import org.apache.logging.log4j.util.MessageSupplier;
+import org.apache.logging.log4j.util.Supplier;
+
+/**
+ * Custom Logger interface with convenience methods for
+ * the OFF, FATAL, ERROR, WARN, INFO, DEBUG, TRACE and ALL custom log levels.
+ * <p>Compatible with Log4j 2.6 or higher.</p>
+ */
+public class ReconciliationLogger implements Serializable {
```

Comment:
Why do we need the `*Op` and `*Cr` methods? On a first look it looks like they might not need the suffixes? I would be curious for the logic behind it. Also ... do we really need all the methods? It looks like some of them will not be used anywhere in out code anyway. I wonder if a class like this used in every class like this changes the memory footprint of the operator.

Reply:
I was inspired by @tombentley 's idea with the suffixes. It is possible to distinguish Cr vs Op logger as CR has the first parameter `Reconciliation` object. So it is possible to have these methods without suffixes IMO.

Reply:
Yeah, that is why it seemed unnecessary to me. So I wondered what the motivation was.